packages/nimble-components/src/table-column/number-text/tests/table-column-number-text.spec.ts

```diff
@@ -0,0 +1,480 @@
+
+                    expect(pageObject.getRenderedCellContent(0, 0)).toBe(
+                        testCase.expectedRenderedString
+                    );
+                    expect(
+                        pageObject.getRenderedGroupHeaderContent(0)
+                    ).toContain(testCase.expectedRenderedString);
+                }
+            );
+        }
+    });
+
+    describe('with integer formatting', () => {
+        beforeEach(async () => {
+            columnInstances.column1.format = NumberTextFormat.integer;
+            await waitForUpdatesAsync();
+        });
+
+        const testCases: readonly NumericTestCase[] = [
+            {
+                name: 'NEGATIVE_INFINITY renders as -∞',
+                value: Number.NEGATIVE_INFINITY,
+                expectedRenderedString: '-∞'
+            },
+            {
+                name: 'POSITIVE_INFINITY renders as ∞',
+                value: Number.POSITIVE_INFINITY,
+                expectedRenderedString: '∞'
+            },
+            {
+                name: 'NaN renders as NaN',
+                value: Number.NaN,
+                expectedRenderedString: 'NaN'
+            },
+            {
+                name: '-0 renders with negative sign',
+                value: -0,
+                expectedRenderedString: '-0'
+            },
+            {
+                name: '+0 renders without positive sign',
+                value: 0,
+                expectedRenderedString: '0'
+            },
+            {
+                name: 'rounds down positive numbers',
+                value: 1.23,
+                expectedRenderedString: '1'
+            },
+            {
+                name: 'rounds up positive numbers',
+                value: 1.76,
+                expectedRenderedString: '2'
+            },
+            {
+                name: 'rounds down negative numbers',
+                value: -1.76,
+                expectedRenderedString: '-2'
+            },
+            {
+                name: 'rounds up negative numbers',
+                value: -1.23,
+                expectedRenderedString: '-1'
+            },
+            {
+                name: 'shows more than 6 digits for positive numbers',
+                value: 987654321,
+                expectedRenderedString: '987,654,321'
+            },
+            {
+                name: 'shows more than 6 digits for negative numbers',
+                value: -123456789,
+                expectedRenderedString: '-123,456,789'
+            },
+            {
+                name: 'MAX_SAFE_INTEGER + 9999 renders as expected',
+                value: Number.MAX_SAFE_INTEGER + 9999,
+                expectedRenderedString: '9,007,199,254,750,990'
+            },
+            {
+                name: 'MIN_SAFE_INTEGER - 9999 renders as expected',
+                value: Number.MIN_SAFE_INTEGER - 9999,
+                expectedRenderedString: '-9,007,199,254,750,990'
+            }
```

Comment:
Maybe split into four tests?

i.e. for the positive cases split to the two tests

```ts
            {
                name: 'MAX_SAFE_INTEGER +1 renders as the exact value',
                value: Number.MAX_SAFE_INTEGER + 1,
                expectedRenderedString: '9,007,199,254,740,992'
            },
            {
                name: 'MAX_SAFE_INTEGER + 1.5  renders with rounding precision errors',
                value: Number.MAX_SAFE_INTEGER + 1.5,
                expectedRenderedString: '9,007,199,254,750,992'
            },
```